File: Materials/SplitApplyCombine_MapReduce/mapper.R
#!/usr/bin/env Rscript
library(stringr)
library(magrittr)

f <- file("stdin")
open(f)
while(length(line <- readLines(f,n=1)) > 0) {
  splitline <- line %>%
    str_to_lower %>%
    str_replace_all("[[:punct:]]", "") %>%
    str_replace_all("\\s\\s+","\\s")   %>%
    str_split("\\s",simplify=FALSE)     %>%
    unlist
  for (word in splitline) {
    cat(word,"\t",1,"\n")
  }
}

File: Materials/SplitApplyCombine_MapReduce/mapper.py
#!/usr/bin/env python
import sys, string

# Read lines from the "standard input" (sys.stdin).
#    sys.stdin is treated like a file object would be.
#    Unless you change it, this means "the console."
#    This could be from your keyboard or the output of
#    a command like "cat."

for line in sys.stdin:

    # lower-case, remove punctuation, strip extra whitespace
    line = line.lower()
    line = line.translate(str.maketrans('','',string.punctuation))
    line = line.strip()

    # split line into words (on spaces)
    words = line.split()

    # print (to standard output) key value pairs: "word 1"
    for word in words:
        print(word,1,sep="\t")
